content/go/event_log.go.md

```diff
@@ -0,0 +1,205 @@
+date: "2021-05-04"
+classes: 
+    - "SoftLayer_Event_Log"
+tags:
+    - "eventlogs"
+    - "objectfilter"
+    - "objectmask"
+    - "resultlimit"
+---
+
+
+This example deals with a few ways of pulling data from [SoftLayer_Event_Log](https://sldn.softlayer.com/reference/services/SoftLayer_Event_Log/). There can be quite a few Logs here, therefore it is recommended to use a filter like in `getRecentLogs` to limit how far back your search for Events, otherwise you will be paging through Events for a long time.
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+	"time"
+)
+
+func main() {
+	sess := session.New()
+	sess.Debug = true
+	getSystemLogs(sess)
+	getLoginLogs(sess)
+	getRecentLogs(sess)
+	getEventUserTypes(sess)
+	getEventNames(sess)
+
+}
+
+/**
+Shows the SYSTEM Logs.
+Request URL:
+GET https://api.softlayer.com/rest/v3.1/SoftLayer_Event_Log/getAllObjects.json?
+objectFilter={"userType":{"operation":"SYSTEM"}}&
+objectMask=mask[userType,eventName,eventCreateDate]&
+resultLimit=0,50
+ */
+func getSystemLogs(sess *session.Session){
+	filter := filter.Build(filter.Path("userType").Eq("SYSTEM"))
+	mask:="userType,eventName,eventCreateDate"
+	systemEvents:=getAllEvents(sess,filter,mask)
+	printLogs(systemEvents)
+}
+
+/**
+Shows the Login Logs
+RequestURL:
+GET https://api.softlayer.com/rest/v3.1/SoftLayer_Event_Log/getAllObjects.json?
+objectFilter={"eventName":{"operation":"^=+Login"}}&
+objectMask=mask[userType;eventName,eventCreateDate]&
+resultLimit=0,50
+ */
+func getLoginLogs(sess *session.Session){
+	filter := filter.Build(filter.Path("eventName").Eq("^= Login"))
+	mask:="userType;eventName,eventCreateDate"
+	eventLogs :=getAllEvents(sess,filter,mask)
+	printLogs(eventLogs)
+}
+
+/**
+Shows the recent Logs
+Request URL:
+GET https://api.softlayer.com/rest/v3.1/SoftLayer_Event_Log/getAllObjects.json?
+objectFilter={"eventCreateDate":{"operation":"greaterThanDate",
+	"options":[{"name":"date","value":["2021-04-24T00:00:00.0000-06:00"]}]}}&
+objectMask=mask[userType;eventName,eventCreateDate]&
+resultLimit=0,50
+ */
+func getRecentLogs(sess *session.Session){
+	offsetDate :=getDateString(-10)
+	filterObject:=filter.Build(filter.Path("eventCreateDate").DateAfter(offsetDate))
+	mask:="userType;eventName,eventCreateDate"
+	recentEvents:=getAllEvents(sess, filterObject,mask)
+	printLogs(recentEvents)
+}
+
+/**
+Pages through all results from the Event_Log. This might take long time.
+*/
+func getAllEvents(
+	sess *session.Session,
+	filter string,
+	mask string,
+) (resp []datatypes.Event_Log) {
+	limit:=50
+	offset:=0
+	eventsSize:=limit
+	var allEvents []datatypes.Event_Log
+	for limit==eventsSize {
+		events := getAllObjects(sess, limit,offset,filter,mask)
+		allEvents=append(allEvents, events...)
+		eventsSize=len(events)
```

Comment:
this should likely be `eventSize = eventSize + len(events)`
and remove the eventSize=0 bit I think.

Reply:
`maxNumberOfEvents ` added to set a max limit.